pandas/_libs/tslibs/parsing.pyx

```diff
@@ -175,19 +181,28 @@ cdef parse_datetime_string_with_reso(date_string, freq=None, dayfirst=False,
 
 
 cpdef bint _does_string_look_like_datetime(object date_string):
-    if date_string.startswith('0'):
-        # Strings starting with 0 are more consistent with a
-        # date-like string than a number
-        return True
-
-    try:
-        if float(date_string) < 1000:
+    cdef:
+        const char *buf
+        char *endptr = NULL
+        Py_ssize_t length = -1
+        double converted_date
+        char first
+        int error = 0
+
+    buf = get_c_string_buf_and_size(date_string, &length)
```

Comment:
give more comments here on what you are doing

Reply:
Comments are given